src/Kiota.Builder/Writers/Java/JavaConventionService.cs

```diff
@@ -53,9 +57,22 @@ public override string GetTypeString(CodeTypeBase code)
             }
             else throw new InvalidOperationException($"type of type {code.GetType()} is unknown");
         }
+        private static readonly CodeUsingDeclarationNameComparer usingDeclarationComparer = new();
+        private static bool IsSymbolDuplicated(string symbol, CodeElement targetElement) {
+            var targetClass = targetElement as CodeClass ?? targetElement.GetImmediateParentOfType<CodeClass>();
+            if (targetClass.Parent is CodeClass parentClass) 
+                targetClass = parentClass;
+            return (targetClass.StartBlock as CodeClass.Declaration)
+                            ?.Usings
+                            ?.Where(x => !x.IsExternal && symbol.Equals(x.Declaration.TypeDefinition.Name, StringComparison.OrdinalIgnoreCase))
+                            ?.Distinct(usingDeclarationComparer)
+                            ?.Count() > 1;
+        }
         public override string TranslateType(CodeType type) {
-            return (type.Name) switch {//TODO we're probably missing a bunch of type mappings
-                ("void" or "boolean") when !type.IsNullable => type.Name.ToFirstCharacterLowerCase(), //little casing hack
+            return type.Name switch {
+                "int64" => "Long",
+                "void" or "boolean" when !type.IsNullable => type.Name.ToFirstCharacterLowerCase(), //little casing hack
```

Comment:
Neither case needs the `ToFirstCharacterLowerCase` call

Reply:
```suggestion
                "void" or "boolean" when !type.IsNullable => type.Name, //little casing hack
```